src/services/expenses/expenses.js

```diff
@@ -0,0 +1,99 @@
+'use strict';
+
+const { v4: uuid } = require('uuid');
+
+const { db } = require('../../db');
+
+const { validateExpenseData } = require('./validation');
+
+const resetStore = async () => {
+  await db`DROP TABLE IF EXISTS expenses`;
+
+  await db`CREATE TABLE expenses (
+    id UUID PRIMARY KEY,
+    userId UUID,
+    title TEXT,
+    category TEXT,
+    amount NUMERIC,
+    spentAt TIMESTAMP,
+    note TEXT
+  )`;
+};
+
+const getAll = async ({ userId, categories, from, to }) => {
+  return db`
+    SELECT *
+    FROM expenses
+    WHERE
+      userId = ${userId}
+      AND
+      spentAt BETWEEN ${from} AND ${to}
+      AND
+      category IN (${categories.join(', ')})
```

Comment:
It will work as long as you and 5 friends will use this app.
Otherwise, you need DB indices.